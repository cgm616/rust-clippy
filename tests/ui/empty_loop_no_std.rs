// ignore-macos
// ignore-windows

#![warn(clippy::empty_loop)]
#![feature(lang_items, link_args, start, libc)]
#![link_args = "-nostartfiles"]
#![no_std]

use core::panic::PanicInfo;

#[start]
fn main(argc: isize, argv: *const *const u8) -> isize {
    // This loop should not cause a warning.
    let dummy = 0u8;
    loop {
        unsafe { core::ptr::read_volatile(&dummy) };
    }
}

#[panic_handler]
fn panic(_info: &PanicInfo) -> ! {
    // But this loop will cause a warning.
    loop {}
}

#[lang = "eh_personality"]
extern "C" fn eh_personality() {}
