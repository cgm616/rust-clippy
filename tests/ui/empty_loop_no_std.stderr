error: empty `loop {}` is currently miscompiled and wastes CPU cycles
  --> $DIR/empty_loop_no_std.rs:23:5
   |
LL |     loop {}
   |     ^^^^^^^
   |
   = note: `-D clippy::empty-loop` implied by `-D warnings`
   = help: You should either use `panic!()` or add some call to sleep or pause the thread to the loop body.

error: aborting due to previous error

